Resolves: Same mesh in several instances of BVH tree

lberti opened the pull request:
- closes #2071

lberti committed:
fix #2071 same mesh for several bvh tree instances

prudhomm commented on feelpp/feel/feelmesh/bvh.hpp:
```diff
@@ -44,6 +44,7 @@ namespace Feel
     template<int nDim>
     class BVHTree
     {
+        typedef BVHTree<nDim> self_type;
```
Please use the readable with `using`
`typedef` is less readable

prudhomm commented on feelpp/feel/feelmesh/bvh.hpp:
```diff
@@ -59,16 +60,18 @@ namespace Feel
 
-        static inline std::vector<BVHPrimitiveInfo> M_primitiveInfo;
-        static inline trace_mesh_ptrtype M_mesh;   
+        // static inline std::vector<BVHPrimitiveInfo> M_primitiveInfo;
```
Remove the commented code if not used anymore

lberti committed:
fix comments PR #2071

lberti committed:
#2071 skip toolboxes skip mor

lberti committed:
Merge branch 'develop' into 2071-same-mesh-in-several-instances-of-bvh-tree

lberti committed:
Merge branch 'develop' into 2071-same-mesh-in-several-instances-of-bvh-tree

prudhomm committed:
update #2071

increase rtol and make it generic

skip feelpp skip toolboxes skip mor skip python
